lib/fileio/translators/translatorMayaReference.cpp

```diff
@@ -0,0 +1,489 @@
+        MPlug srcPlug,
+        MPlug destArrayPlug,
+        unsigned int& foundIndex,
+        bool& wasConnected)
+    {
+        // Want to find the lowest unconnected (as dest) open logical index... so we add to
+        // a list, then sort
+        MStatus status;
+        foundIndex = 0;
+        wasConnected = false;
+        unsigned int numConnected = destArrayPlug.numConnectedElements(&status);
+        CHECK_MSTATUS_AND_RETURN_IT(status);
+        if (numConnected > 0)
+        {
+            std::vector<unsigned int> usedLogicalIndices;
+            usedLogicalIndices.reserve(numConnected);
+            MPlug elemPlug;
+            MPlug elemSrcPlug;
+            for (unsigned int connectedI = 0; connectedI < numConnected; ++connectedI)
+            {
+                elemPlug = destArrayPlug.connectionByPhysicalIndex(connectedI, &status);
+                CHECK_MSTATUS_AND_RETURN_IT(status);
+                elemSrcPlug = elemPlug.source(&status);
+                CHECK_MSTATUS_AND_RETURN_IT(status);
+                if (!elemSrcPlug.isNull())
+                {
+                    if (elemSrcPlug == srcPlug)
+                    {
+                        foundIndex = elemPlug.logicalIndex();
+                        wasConnected = true;
+                        return status;
+                    }
+                    usedLogicalIndices.push_back(elemPlug.logicalIndex());
+                }
+            }
+            if (!usedLogicalIndices.empty())
+            {
+                std::sort(usedLogicalIndices.begin(), usedLogicalIndices.end());
+                // after sorting, since we assume no repeated indices, if the number of
+                // elements = value of last element + 1, then we know it's tightly packed...
+                if (usedLogicalIndices.size() - 1 == usedLogicalIndices.back())
+                {
+                    foundIndex = usedLogicalIndices.size();
+                }
+                else
+                {
+                    // If it's not tightly packed, just iterate through from start until we
+                    // find an element whose index != it's value
+                    for (foundIndex = 0;
+                        foundIndex < usedLogicalIndices.size();
+                        ++foundIndex)
+                    {
+                        if (usedLogicalIndices[foundIndex] != foundIndex) break;
+                    }
+                }
+            }
+        }
+        return status;
+    }
+
+  // Given a function set attached to a DAG node, create a string from the
+  // node's full path by replacing all the path dividers (|) by underscores
+  // and stripping off the first character so the name doesn't start with an
+  // underscore.  (We use this for creating unique names for Maya reference
+  // nodes created from ALUSD proxy nodes.)
+  //
+  MString refNameFromPath(const MFnDagNode &nodeFn)
+  {
+    MString name = nodeFn.fullPathName();
+    name.substitute("|", "_");
```

Comment:
You are replacing single characters. Calling a method designed for strings is overkill if you ask me. 
```c++
std::replace(name.asChar(), name.asChar() + name.length(), '|', '_');
```